src/components/common/Extensions/UserHub/partials/ReputationTab/partials/PendingReputation/PendingReputation.tsx

```diff
@@ -28,9 +30,30 @@ const PendingReputation: FC<PendingReputationProps> = ({
 
   const { data } = useGetReputationMiningCycleMetadataQuery();
   const { lastCompletedAt } = data?.getReputationMiningCycleMetadata ?? {};
-  const nextMiningCycleDate = lastCompletedAt
-    ? getNextMiningCycleDate(new Date(lastCompletedAt ?? ''))
-    : null;
+
+  const [nextMiningCycleDate, setNextMiningCyleDate] = useState(
+    lastCompletedAt
+      ? getNextMiningCycleDate(new Date(lastCompletedAt ?? ''))
+      : null,
+  );
+
+  useEffect(() => {
+    const intervalTimer = setInterval(() => {
+      if (!nextMiningCycleDate) {
+        return;
+      }
+
+      const halfIntervalAgo = new Date(
+        Date.now() - (UPDATE_INTERVAL / 2) * 1000,
+      );
+
+      if (nextMiningCycleDate < halfIntervalAgo) {
+        setNextMiningCyleDate(getNextMiningCycleDate(halfIntervalAgo));
```

Comment:
Should we use `halfIntervalAgo` for the next mining cycle as opposed to `nextMiningCycleDate`?

Or maybe we could refetch the query, if that would return the updated cycle date?

Reply:
Ah yeah, on reflection, makes more sense to use nextMiningCycleDate to set the next one.

We could refetch the query, but my worry is as this is just an estimate, when we refetch it is possible that the next mining cycle won't have actually ran yet and we'd have to keep calling the query until it has updated. I can look at doing that though if that is preferred?

Reply:
That's a good point, let's leave it like it is. Nice one!